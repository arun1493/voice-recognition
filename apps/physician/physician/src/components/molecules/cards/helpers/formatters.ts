import i18n from "i18next";
import _get from "lodash/get";
import { EMPTY_OBJECT } from "@suki-web/suki-business";

import {
  Allergy,
  AllergyReaction,
  CardMedicalHistoryProps,
  CardSurgicalHistoryProps,
  ParagraphCard,
  Problem
} from "../../../../types";
import { CardType } from "../../../../constants";

const formatParagraphCardData = (data: CardMedicalHistoryProps | CardSurgicalHistoryProps | any, cardType: string) => {
  const paragraphData: Array<ParagraphCard> = [];

  switch (cardType) {
    case CardType.ALLERGIES: {
      const { allergies, nkda } = data || EMPTY_OBJECT;

      allergies &&
        allergies.forEach((allergyItem: Allergy) => {
          const { allergen_name, reactions } = allergyItem || EMPTY_OBJECT;
          let allergyReactionContent = "";
          reactions &&
            reactions.forEach((allergyItem: AllergyReaction) => {
              const { reaction_name } = allergyItem;
              if (reaction_name) allergyReactionContent = allergyReactionContent.concat(reaction_name, ", ");
            });

          allergyReactionContent = allergyReactionContent.slice(0, -2); // to remove comma at the end of string

          if (allergen_name)
            paragraphData.push({
              title: allergen_name,
              content: allergyReactionContent
            });
        });

      if (nkda)
        paragraphData.push({
          content: i18n.t("pages.note.cards.errors.allergiesNkda")
        });
      break;
    }
    case CardType.PAST_MEDICAL_HISTORY: {
      const { active_problems } = data || EMPTY_OBJECT;
      active_problems &&
        active_problems.forEach((activeProblem: Problem) => {
          const description = _get(activeProblem, "note");
          const error = _get(activeProblem, "error");
          const title = _get(activeProblem, "diagnosis.icd_description");
          if (!error) paragraphData.push({ title, content: description });
        });
      break;
    }

    case CardType.PAST_SURGICAL_HISTORY: {
      data &&
        data.forEach((surgicalHistoryItem: Problem) => {
          const { description, note, error } = surgicalHistoryItem;
          if (!error) paragraphData.push({ title: description, content: note });
        });
      break;
    }

    default:
      break;
  }
  return paragraphData;
};

export { formatParagraphCardData };
