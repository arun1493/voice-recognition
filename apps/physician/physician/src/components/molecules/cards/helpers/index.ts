export * from "./labels";
export * from "./formatters";
