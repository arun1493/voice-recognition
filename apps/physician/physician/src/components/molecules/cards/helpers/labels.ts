import i18n from "i18next";

import { CardType, VITAL_SIGNS } from "../../../../constants";

const getCardTitle = (cardType: string) => {
  switch (cardType) {
    case CardType.ALLERGIES:
      return i18n.t("pages.note.cards.titles.allergies");
    case CardType.CURRENT_MEDICATION:
      return i18n.t("pages.note.cards.titles.medication");
    case CardType.PAST_MEDICAL_HISTORY:
    case CardType.PAST_MEDICAL_HISTORY_SHOW_ME:
      return i18n.t("pages.note.cards.titles.medicalHistory");
    case CardType.PAST_SURGICAL_HISTORY:
    case CardType.PAST_SURGICAL_HISTORY_SHOW_ME:
      return i18n.t("pages.note.cards.titles.surgicalHistory");
    case CardType.VITALS:
      return i18n.t("pages.note.cards.titles.vitals");
    default:
      return "";
  }
};

const getVitalSign = (sign: string) => {
  switch (sign) {
    case VITAL_SIGNS.BLOOD_PRESSURE:
      return i18n.t("pages.note.cards.vitalSigns.bloodPressure");
    case VITAL_SIGNS.BMI:
      return i18n.t("pages.note.cards.vitalSigns.bmi");
    case VITAL_SIGNS.HEIGHT:
      return i18n.t("pages.note.cards.vitalSigns.height");
    case VITAL_SIGNS.PULSE:
      return i18n.t("pages.note.cards.vitalSigns.pulse");
    case VITAL_SIGNS.WEIGHT:
      return i18n.t("pages.note.cards.vitalSigns.weight");
    case VITAL_SIGNS.HEART_RATE:
      return i18n.t("pages.note.cards.vitalSigns.heartRate");
    case VITAL_SIGNS.RESPIRATORY_RATE:
      return i18n.t("pages.note.cards.vitalSigns.respiratoryRate");
    case VITAL_SIGNS.TEMPERATURE:
      return i18n.t("pages.note.cards.vitalSigns.temperature");
    default:
      return null;
  }
};

const getErrorMessage = (cardType: string) => {
  switch (cardType) {
    case CardType.ALLERGIES:
      return i18n.t("pages.note.cards.errors.allergies");
    case CardType.CURRENT_MEDICATION:
      return i18n.t("pages.note.cards.errors.medication");
    case CardType.PAST_MEDICAL_HISTORY:
    case CardType.PAST_MEDICAL_HISTORY_SHOW_ME:
      return i18n.t("pages.note.cards.errors.medicalHistory");
    case CardType.PAST_SURGICAL_HISTORY:
    case CardType.PAST_SURGICAL_HISTORY_SHOW_ME:
      return i18n.t("pages.note.cards.errors.surgicalHistory");
    case CardType.VITALS:
      return i18n.t("pages.note.cards.errors.vitals");
    default:
      return "";
  }
};

export { getCardTitle, getErrorMessage, getVitalSign };
