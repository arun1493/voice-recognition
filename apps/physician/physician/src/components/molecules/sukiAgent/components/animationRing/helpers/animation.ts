import { RING_ANIMATION_DURATION, RING_BASE_OPACITY, RING_SCALE, SukiAgentState } from "../../../../../../constants";

function isValidSukiAgentStateForSingleRingAnimation(agentStatus: SukiAgentState, isAgentActive: boolean) {
  return (isAgentActive && agentStatus === SukiAgentState.AWAKE) || agentStatus === SukiAgentState.PROCESSING;
}

function isValidSukiAgentStateForMultipleRingAnimation(agentStatus: SukiAgentState, isAgentActive: boolean) {
  return (
    isAgentActive &&
    !isValidSukiAgentStateForSingleRingAnimation(agentStatus, isAgentActive) &&
    agentStatus === SukiAgentState.LISTENING
  );
}

function getDictationRingVariants(ringsCount: number) {
  const dictationRingVariants = [];
  for (let i = ringsCount; i > 0; i--) {
    const RING_VARIANTS = {
      oscillate: {
        opacity: RING_BASE_OPACITY - i * 0.2,
        scale: RING_SCALE + i * 0.01,
        transition: {
          yoyo: Infinity,
          duration: RING_ANIMATION_DURATION,
          type: "tween",
          delay: i * 0.1,
          ease: "easeInOut"
        }
      }
    };

    dictationRingVariants.push(RING_VARIANTS);
  }
  return dictationRingVariants;
}

export {
  getDictationRingVariants,
  isValidSukiAgentStateForMultipleRingAnimation,
  isValidSukiAgentStateForSingleRingAnimation
};
