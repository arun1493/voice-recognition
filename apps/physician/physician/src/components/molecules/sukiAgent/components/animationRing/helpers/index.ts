export * from "./ringSize";
export * from "./animation";
