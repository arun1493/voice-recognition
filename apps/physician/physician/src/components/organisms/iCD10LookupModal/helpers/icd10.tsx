import _isEmpty from "lodash/isEmpty";

const formatText = (text: string, highlight: string) => {
  if (_isEmpty(text)) return;
  const parts = text.split(new RegExp(`(${highlight})`, "gi"));
  return parts.map((part) => (part.toLowerCase() === highlight.toLowerCase() ? <strong>{part}</strong> : part));
};

const formatCode = (code: string) => {
  if (code?.length > 3) {
    return code?.substring(0, 3) + "." + code?.substring(3);
  }
  return code;
};

export { formatCode, formatText };
