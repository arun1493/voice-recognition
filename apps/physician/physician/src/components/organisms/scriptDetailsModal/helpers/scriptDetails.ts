import { ScreenSize } from "@suki-web/suki-ui";

const { SM, XS, XXS } = ScreenSize;

const getWidth = (breakPoint: ScreenSize) => {
  switch (breakPoint) {
    case XXS:
      return "100vw";
    case XS:
      return 484;
    case SM:
      return 664;
    default:
      return 782;
  }
};

const getHeight = (breakPoint: ScreenSize) =>
  breakPoint === XS || breakPoint === XXS ? "100vh" : "calc( 100vh - 56px )";

const getMaxHeight = (breakPoint: ScreenSize) => (breakPoint === XS || breakPoint === XXS ? "100vh" : 0);

export { getHeight, getMaxHeight, getWidth };
