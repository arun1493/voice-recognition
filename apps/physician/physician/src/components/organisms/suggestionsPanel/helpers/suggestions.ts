import i18n from "i18next";

import { SuggestionTypes } from "../../../../constants";

const getTextBySuggestion = (suggestion: string) => {
  switch (suggestion) {
    case SuggestionTypes.PL:
      return i18n.t("components.suggestionsPanel.pl");
    case SuggestionTypes.ED:
      return i18n.t("components.suggestionsPanel.ed");
    default:
      return "";
  }
};

export { getTextBySuggestion };
