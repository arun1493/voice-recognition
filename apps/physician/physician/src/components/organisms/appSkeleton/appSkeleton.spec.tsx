import { cleanup, RenderResult } from "@testing-library/react";

import { AppSkeleton } from ".";
import { configureI18n } from "../../../helpers";
import { componentSetupWithStore } from "../../../hocs";
import { shouldMatchSnapshot } from "../../../helpers/snapshot.spec";

const setup = componentSetupWithStore(() => <AppSkeleton />);

describe("AppSkeleton", () => {
  let component: RenderResult;

  beforeEach(() => {
    configureI18n();
    component = setup();
  });

  afterEach(() => {
    cleanup();
  });

  it("should render successfully", () => {
    const { baseElement } = component;
    expect(baseElement).toBeTruthy();
  });

  it("should match snapshot", () => {
    shouldMatchSnapshot(AppSkeleton);
  });
});
