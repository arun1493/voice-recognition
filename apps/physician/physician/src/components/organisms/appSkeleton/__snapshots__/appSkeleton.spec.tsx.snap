// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppSkeleton should match snapshot 1`] = `
<div
  className="appSkeletonstyled__Container-sc-1q0blj0-0 evWHTF"
>
  <header
    className="appSkeletonstyled__Header-sc-1q0blj0-3 jFKDoV"
  >
    <a
      className="SukiLogo__StyledLink-empixj-0 eltBpC"
      data-cy="suki-logo"
      height={24}
      href="/home"
      onClick={[Function]}
      width={74}
    >
      <svg
        viewBox="0 0 117, 37"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          fill="currentColor"
          fillRule="nonzero"
        >
          <path
            d="M0 13.272C0 6.254 4.55 2.146 17.104 2.146c7.382 0 12.327 1.28 14.45 2.559V11.9c-2.431-1.014-8.306-2.558-14.85-2.558-4.729 0-8.263.971-8.263 3.665 0 2.65 1.902 3.045 7.647 3.4 12.418.751 16.44 2.873 16.44 9.755 0 7.86-5.967 10.77-17.148 10.77-6.144 0-12.11-1.15-14.98-2.516v-6.974c2.475 1.058 8.041 2.472 14.407 2.472 6.54 0 9.236-1.101 9.236-3.622 0-1.9-1.195-2.694-8.22-3.136C4.463 22.454 0 20.246 0 13.272ZM36.824 10.504h8.116v14.812c0 3.518 1.898 4.707 5.602 4.707 3.705 0 6.525-1.054 8.817-2.33v-17.19h8.067v25.81h-7.673v-2.991c-2.82 1.845-6.568 3.61-11.6 3.61-7.802 0-11.334-3.476-11.334-9.416V10.504h.005ZM73.073 36.367V1.581h8.097v19.224l11.642-10.377h10.933L90.111 21.334l14.473 15.033h-11.02l-8.941-10.436-3.453 2.92v7.516zM107.52 3.749c0-2.5 1.496-3.749 4.666-3.749 3.212 0 4.708 1.25 4.708 3.749 0 2.455-1.496 3.705-4.708 3.705-3.17-.005-4.666-1.255-4.666-3.705ZM108.311 10.391h7.793v25.977h-7.793z"
          />
        </g>
      </svg>
    </a>
    <nav
      className="navigationToolbarstyled__StyledNavigationToolbar-sc-1qo6xy4-0 eaMgJk"
      data-cy="navigation-toolbar"
    >
      <div
        className="sukiTooltipstyled__StyledSukiTooltipWrapper-sc-175s0ct-0 hlZCBe"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <button
          className="iconButtonstyled__StyledIconButton-v3k1vq-0 gtCISe"
          data-cy="navigation-toolbar-search"
          onClick={[Function]}
          size="md"
          type="button"
        >
          <i
            className="SukiIcon__StyledIcon-q9b9ar-0 aoDzN sukiIcon "
            color="inherit"
            size="sm"
          >
            <svg
              fill="none"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9.842 3.043a6.191 6.191 0 0 0-2.022.548c-1.972.898-3.326 2.651-3.73 4.829-.092.495-.092 1.665 0 2.16.514 2.771 2.559 4.816 5.33 5.33.495.092 1.665.092 2.16 0 2.771-.514 4.816-2.559 5.33-5.33.092-.495.092-1.665 0-2.16-.451-2.433-2.09-4.326-4.41-5.095a6.485 6.485 0 0 0-2.658-.282m1.218 2.001a4.52 4.52 0 0 1 3.48 2.493c.721 1.489.559 3.304-.411 4.606-.582.782-1.313 1.319-2.2 1.616a4.4 4.4 0 0 1-2.858 0c-1.981-.664-3.253-2.632-3.038-4.699a4.484 4.484 0 0 1 5.027-4.016m3.86 10.471c-.3.098-.554.326-.671.602-.076.181-.075.542.002.729.065.154 2.931 4.263 3.099 4.442.252.269.742.359 1.096.202.217-.096.423-.298.519-.509.09-.197.097-.57.016-.765-.069-.165-2.893-4.212-3.077-4.41a1.26 1.26 0 0 0-.281-.214c-.172-.088-.544-.129-.703-.077"
                fill="currentColor"
                fillRule="evenodd"
              />
            </svg>
          </i>
        </button>
      </div>
      <div
        className="sukiTooltipstyled__StyledSukiTooltipWrapper-sc-175s0ct-0 hlZCBe"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <button
          className="iconButtonstyled__StyledIconButton-v3k1vq-0 gtCISe"
          data-cy="patients-list-icon"
          onClick={[Function]}
          size="md"
          type="button"
        >
          <i
            className="SukiIcon__StyledIcon-q9b9ar-0 aoDzN sukiIcon "
            color="inherit"
            size="sm"
          >
            <svg
              fill="none"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9.4 2.044c-1.639.185-3.139 1.246-3.874 2.741C5.158 5.533 5.02 6.136 5.02 7c0 .67.058 1.044.248 1.6.578 1.691 1.985 2.925 3.772 3.307.471.101 1.449.101 1.92 0 1.528-.327 2.788-1.277 3.473-2.619.405-.794.547-1.389.547-2.288 0-.899-.142-1.494-.547-2.287a5.022 5.022 0 0 0-3.193-2.551 5.223 5.223 0 0 0-1.84-.118m1.461 2.085c.943.276 1.723 1.054 2.014 2.01.083.273.098.403.098.861 0 .458-.015.588-.098.861a3.073 3.073 0 0 1-2.035 2.025c-.434.135-1.246.135-1.68 0a3.073 3.073 0 0 1-2.035-2.025c-.083-.273-.098-.403-.098-.861 0-.458.015-.588.098-.861A3.042 3.042 0 0 1 9.499 4.04c.333-.06 1.001-.017 1.362.089m4.993.411c.013.033.069.227.124.43.45 1.664.304 3.456-.402 4.937-.353.739-.945 1.491-1.513 1.921l-.243.183.4-.023a4.01 4.01 0 0 0 3.497-2.515c.511-1.275.304-2.773-.535-3.889a3.783 3.783 0 0 0-.977-.909c-.309-.204-.389-.235-.351-.135M7.2 13.059a5.7 5.7 0 0 0-1.88.581c-1.655.831-2.791 2.31-3.205 4.169-.104.47-.155 2.159-.077 2.562a2.05 2.05 0 0 0 1.554 1.585c.299.062 12.517.062 12.816 0a2.05 2.05 0 0 0 1.553-1.583c.08-.409.031-2.055-.075-2.547-.533-2.458-2.344-4.235-4.826-4.734-.443-.089-5.269-.116-5.86-.033m8.38.08c1.556.462 2.822 1.548 3.59 3.081.523 1.044.67 1.882.67 3.833v.953l.71-.013c.8-.015.903-.043 1.176-.324.258-.265.286-.423.264-1.463-.018-.854-.026-.933-.142-1.366-.274-1.021-.711-1.805-1.423-2.549-1.101-1.152-2.56-1.861-4.465-2.17-.651-.106-.777-.1-.38.018m-2.622 1.975a4.277 4.277 0 0 1 1.865 1.063c.342.342.569.671.801 1.16.276.584.34.908.365 1.853l.021.81H3.99l.021-.81c.025-.945.089-1.269.365-1.853.232-.489.459-.818.801-1.16a4.098 4.098 0 0 1 2.256-1.135c.125-.019 1.343-.031 2.707-.026 2.421.009 2.488.011 2.818.098"
                fill="currentColor"
                fillRule="evenodd"
              />
            </svg>
          </i>
        </button>
      </div>
      <div
        className="mainMenuNavigationstyled__MainAndDropDownMenuContainer-sc-15amjwi-0 kSHPqG"
      >
        <div
          className="sukiTooltipstyled__StyledSukiTooltipWrapper-sc-175s0ct-0 hlZCBe"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <button
            className="iconButtonstyled__StyledIconButton-v3k1vq-0 gtCISe"
            data-cy="hamburger-menu"
            onClick={[Function]}
            size="md"
            type="button"
          >
            <i
              className="SukiIcon__StyledIcon-q9b9ar-0 aoDzN sukiIcon "
              color="inherit"
              size="sm"
            >
              <svg
                fill="currentColor"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <rect
                  fill="currentColor"
                  height="2"
                  rx="1"
                  width="14"
                  x="5"
                  y="6"
                />
                <rect
                  fill="currentColor"
                  height="2"
                  rx="1"
                  width="14"
                  x="5"
                  y="11"
                />
                <rect
                  fill="currentColor"
                  height="2"
                  rx="1"
                  width="14"
                  x="5"
                  y="16"
                />
              </svg>
            </i>
          </button>
        </div>
      </div>
    </nav>
  </header>
  <div
    className="appSkeletonstyled__Content-sc-1q0blj0-4 cGoulE"
  >
    <button
      className="buttonstyled__StyledButton-ws70ng-0 tGFSu newNoteButtonstyled__StyledNewNoteButton-sc-1q4eklm-1 kLuVnl"
      data-cy="new-note-button"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <div
        className="newNoteButtonstyled__StyledNewNoteButtonContent-sc-1q4eklm-2 dRjnJj"
      >
        <svg
          fill="none"
          height={24}
          viewBox="0 0 24 24"
          width={24}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M13 11C13 10.4477 12.5523 10 12 10C11.4477 10 11 10.4477 11 11V13H9C8.44772 13 8 13.4477 8 14C8 14.5523 8.44772 15 9 15H11V17C11 17.5523 11.4477 18 12 18C12.5523 18 13 17.5523 13 17V15H15C15.5523 15 16 14.5523 16 14C16 13.4477 15.5523 13 15 13H13V11Z"
            fill="#2e5f62"
            fillRule="evenodd"
            opacity="1"
          />
          <path
            clipRule="evenodd"
            d="M5 1H15.1716C15.702 1 16.2107 1.21071 16.5858 1.58579L20.4142 5.41421C20.7893 5.78929 21 6.29799 21 6.82843V21C21 22.1046 20.1046 23 19 23H5C3.89543 23 3 22.1046 3 21V3C3 1.89543 3.89543 1 5 1ZM14 3H6C5.44772 3 5 3.44772 5 4V20C5 20.5523 5.44772 21 6 21H18C18.5523 21 19 20.5523 19 20V8C19 7.44772 18.5523 7 18 7H16C15.4477 7 15 6.55228 15 6V4C15 3.44772 14.5523 3 14 3Z"
            fill="#2e5f62"
            fillRule="evenodd"
            opacity="1"
          />
        </svg>
        <p
          className="textstyled__StyledText-sc-1vedkxi-0 gQoCmW newNoteButtonstyled__StyledNewNoteText-sc-1q4eklm-0 eSkrkz"
          color="inherit"
          cursor="default"
          onClick={[Function]}
          size="md"
        >
          New Note
        </p>
      </div>
    </button>
  </div>
</div>
`;
